internal/plugin/server.go

```diff
@@ -523,3 +520,10 @@ func (plugin *NvidiaDevicePlugin) apiDeviceSpecs(driverRoot string, ids []string
 
 	return specs
 }
+
+func ensureEnvvar(response *pluginapi.ContainerAllocateResponse, key string, value string) {
+	if response.Envs == nil {
+		response.Envs = make(map[string]string)
+	}
+	response.Envs[key] = value
+}
```

Comment:
I don't love that this is called all over the place, forcing us to always do the check for `nil`.

When is it _ever_ `nil` by the time it's in this function (meaning we never did this check before any of the assignment calls before).

Is it already initialized from:
```
response, err := plugin.getAllocateResponseForCDI(responseID, deviceIDs)
```

Reply:
For example if CDI is enabled and GDS as well, there is currently a bug since we don't init the envvars then. We could set Envs to an empty map at the start of the allocate function to remove the need for this.

Reply:
To clarify, we are checking whether the `Envs` map is nil. Since we just instantiate a `ContainerAllocateResponse` object in `getAllocateResponseForCDI` this would still be nil since we never set any envvars there.